tests/foreman/longrun/test_inc_updates.py

```diff
@@ -0,0 +1,338 @@
+            environment=cls.qe_lce.id,
+            host_collection=[cls.rhel_6_partial_hc],
+        ).create()
+
+        # Step 14: Assign subscriptions to activation keys
+        # Fetch available subscriptions
+        subs = Subscription(organization=cls.org).search()
+        assert len(subs) > 0
+
+        # Add subscriptions to activation key
+        print 'Adding subscriptions to both Activation keys'
+        for sub in subs:
+            if sub.read_json()['product_name'] == u'Employee SKU':
+                for act_key in [rhel_6_ak, rhel_6_partial_ak]:
+                    act_key.add_subscriptions(data={
+                        u'subscription_id': sub.id
+                    })
+
+        # Step 15: Enable product content in activation keys
+        for act_key in [rhel_6_ak, rhel_6_partial_ak]:
+            act_key.content_override(data={'content_override': {
+                u'content_label': PRD_SETS['rhel6_sat6tools']['label'],
+                u'value': u'1'
+            }})
+
+        # Step 16: Create client machines
+        cls.vm1 = VirtualMachine(distro='rhel67', tag='incupdate')
+        cls.vm2 = VirtualMachine(distro='rhel67', tag='incupdate')
+        cls.vm3 = VirtualMachine(distro='rhel67', tag='incupdate')
+        cls.setup_vm(cls.vm1, rhel_6_ak.name, cls.org.label)
+        cls.setup_vm(cls.vm2, rhel_6_partial_ak.name, cls.org.label)
+        cls.setup_vm(cls.vm3, rhel_6_partial_ak.name, cls.org.label)
+
+        # Find the content hosts and name them
+        cls.contenthost1, cls.contenthost2, cls.contenthost3 = (None, ) * 3
+        systems = System(organization=cls.org).search()
+        for system in systems:
+            if system.name == cls.vm1.hostname:
+                cls.contenthost1 = system
+            elif system.name == cls.vm2.hostname:
+                cls.contenthost2 = system
+            elif system.name == cls.vm3.hostname:
+                cls.contenthost3 = system
+
+    @classmethod
+    def tearDownClass(cls):
+        """Destroys all provisioned vms"""
+        cls.vm1.destroy()
+        cls.vm2.destroy()
+        cls.vm3.destroy()
+        entity_mixins.TASK_TIMEOUT = cls.old_task_timeout
+
+    @staticmethod
+    def setup_vm(client, act_key, org_name):
+        """Creates the vm and registers it to the satellite"""
+        client.create()
+        client.install_katello_cert()
+
+        # Register content host, install katello-agent
+        result = client.register_contenthost(
+            act_key,
+            org_name,
+            releasever='6.7'
+        )
+        assert result.return_code == 0
+        result = client.install_katello_agent()
+        client.run('katello-package-upload')
+        return
+
+    @staticmethod
+    def get_content_view_versions(content_view_id):
+        """Retrieves content view versions for the given content view id"""
+        return ContentView(id=content_view_id).read_json()['versions']
```

Comment:
What needs to be returned here? IDs? Names?

If IDs need to be returned, we can just do:

```python
return [version.id for version in ContentView(id=content_view_id).read().version]
```

But that makes me wonder whether we need this method at all, or whether the places that call this method could be simplified.